### ISSUE TITLE
Add "label" support to sessions

### ISSUE THREAD

**Issue by OJ** [rapid7/metasploit-framework#8425]
Sessions are currently identifiable by ip, machine name and user context when viewed from the sessions list. It'd be fantastic to have the ability to add some kind of label to the session to make it more identifiable. For example. if I pop SYSTEM on a CFO's desktop, I don't have the ability to know that it's the CFO's desktop from looking at the session list. I'd like to be able to do something like this:
```
meterpreter> sessions -i 3 -label CFO-DESKTOP
```
Then when I run `sessions -x` I can see the label that I have applied. It'd be nice to also be able to switch sessions based on this tag as well (using `sessions CFO-DESKTOP` for example).

Thoughts on this? I'm happy to implement it myself. If people would see this as useful, would it require storing in the DB metadata alongside the rest of the session information?

**Comment by busterb** [comment]
Sounds good to me. It might be even better to tie this to machine_id so if it reconnects you can still see it is CFO-DESKTOP (or rather CFO-LAPTOP, who uses desktops these days :)

**Comment by busterb** [comment]
hey @darkbushido @clee-r7 please give this a gander (they're busy fixing a bunch of stuff in the database now, so making sure they're looped in)

**Comment by OJ** [comment]
Yeah that's a good idea. Would be great to have that persist across sessions. However, it'd be better to tie to the soon_to_be_implemented session ID, because we may have multiple sessions on the same machine with different labels?

**Comment by busterb** [comment]
Oh, you're thinking in terms of _that_ session ID! Got it.
